src/schema/README.md

```diff
@@ -35,3 +35,165 @@ of the schema, roughly in order of importance:
     folder as a BIDS dataset, but which do not follow the BIDS structure
     internally, such as `code` or `sourcedata`. The schema specifies which
     folders are accepted and whether they are required or optional.
+
+-   `metadata/*.yaml`: Valid fields for sidecar metadata json files.
+    These files contain, at minimum, the following fields: `name`,
+    `description`, and a set of fields for describing the field's data type.
+
+    The data types include `type`, which MUST have a value of
+    `array`, `string`, `integer`, `number`, `object`, or `boolean`.
+    There are additional fields which may define rules that apply to a given
+    type.
+
+    -   `array`: If `type` is `array`, then there MUST be an `items` field at
+        the same level as `type`. `items` describes the data type and rules that
+        apply to the individual items in the array. The same rules that apply to
+        describing data types for the field itself apply to `items`.
+        Additionally, there may be any of the following fields at the same level
+        as `type`: `minItems`, `maxItems`.
+        The minimum number of items (`minItems`) defaults to 1.
```

Comment:
so empty arrays are never allowed? Did you check the spec whether that's the case everywhere? If we have a case of a REQUIRED array field and no empty arrays are allowed, we must allow `"n/a"` for that field IMHO.

Reply:
I adopted this in https://github.com/bids-standard/bids-specification/pull/762#discussion_r609063929, but now that I can see the old conversation it's clear that there was no consensus or real basis in the specification behind it.

I like your idea of `"n/a"` over an empty array, but I don't know how users actually do things right now, so any rule we start applying could be backwards incompatible, right? I kind of wish I hadn't removed all of the `minItems: 1`s from the schema in the old PR now...

Reply:
running the following in bids-examples shows some usage of `[]` --> `grep \\[\\] -r . --exclude='*.nii*' -I --exclude-dir={*code,*.git}`



```
./eeg_rishikesh/dataset_description.json:  "Funding": [],
./ieeg_epilepsy_ecog/dataset_description.json: "Funding": [],
./ds003/dataset_description.json:    "Funding": [],
./ds003/dataset_description.json:    "ReferencesAndLinks": [],
./ds000117/sub-13/ses-meg/sub-13_ses-meg_task-facerecognition_meg.json:   "None": []
./ds000117/sub-emptyroom/sub-emptyroom_task-noise_meg.json:   "None": []
./ds000117/sub-04/ses-meg/sub-04_ses-meg_task-facerecognition_meg.json:   "None": []
./ds000117/sub-15/ses-meg/sub-15_ses-meg_task-facerecognition_meg.json:   "None": []
./ds000117/sub-12/ses-meg/sub-12_ses-meg_task-facerecognition_meg.json:   "None": []
./ds000117/sub-16/ses-meg/sub-16_ses-meg_task-facerecognition_meg.json:   "None": []
./ds000117/sub-06/ses-meg/sub-06_ses-meg_task-facerecognition_meg.json:   "None": []
./ds000117/sub-14/ses-meg/sub-14_ses-meg_task-facerecognition_meg.json:   "None": []
./ds000117/sub-02/ses-meg/sub-02_ses-meg_task-facerecognition_meg.json:   "None": []
./ds000117/sub-11/ses-meg/sub-11_ses-meg_task-facerecognition_meg.json:   "None": []
./ds000117/sub-08/ses-meg/sub-08_ses-meg_task-facerecognition_meg.json:   "None": []
./ds000117/derivatives/meg_derivatives/sub-14/ses-meg/sub-14_ses-meg_task-facerecognition_proc-sss_meg.json:   "None": []
./ds000117/sub-07/ses-meg/sub-07_ses-meg_task-facerecognition_meg.json:   "None": []
./ds000117/sub-05/ses-meg/sub-05_ses-meg_task-facerecognition_meg.json:   "None": []
./ds000117/sub-01/ses-meg/sub-01_ses-meg_task-facerecognition_meg.json:   "None": []
./ds000117/sub-10/ses-meg/sub-10_ses-meg_task-facerecognition_meg.json:   "None": []
./ds000117/sub-09/ses-meg/sub-09_ses-meg_task-facerecognition_meg.json:   "None": []
./ds000117/sub-03/ses-meg/sub-03_ses-meg_task-facerecognition_meg.json:   "None": []
```

Reply:
Am I reading those correctly? Are they instances of a metadata field called `"None"` with a value of `[]`? A field named "None" doesn't seem to be in the spec.

Reply:
> Am I reading those correctly? Are they instances of a metadata field called "None" with a value of []? A field named "None" doesn't seem to be in the spec.

Yeah, I was confused as well at first, but it seems legit:

```
 "SoftwareFilters": {
  "SpatialCompensation": {
   "None": []
  }
 }
```

https://github.com/bids-standard/bids-examples/blob/c3fdb3ac3294426d8c245a2ec2419aa0b202e0b5/ds000117/sub-04/ses-meg/sub-04_ses-meg_task-facerecognition_meg.json#L25-L29

That's because there is very little standardization currently on how to specify filters.

Reply:
Okay, I'm just going to (1) drop the mention of the `minItems` default in the README and (2) add back in `minItems: 1` cases into the schema. I removed them in 04120de4aa59b1621242b611a4b9834042e41c2a and it looks like there were only two affected fields: `GeneratedBy` and `VolumeTiming`.

Reply:
okay :+1: we don't need to solve that issue in this PR. Better to leave it for later

Reply:
Should be good as of e5c130e.